merklehasher/inclusion_proof.go

```diff
@@ -0,0 +1,255 @@
+package merklehasher
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+
+	"github.com/pkg/errors"
+
+	"github.com/iotaledger/iota.go/v3"
+)
+
+type hashable interface {
+	Hash(hasher *Hasher) []byte
+}
+
+type leafValue struct {
+	Value []byte
+}
+
+type hashValue struct {
+	Value []byte
+}
+
+type InclusionProof struct {
```

Comment:
I would probable rename `InclusionProof` to just `Proof`. Usually it is called a "Merkle Proof" and that maybe makes it a bit more concise.
But "inclusion proof" is not wrong